src/Puzzle.java

```diff
@@ -7,21 +11,135 @@
 public class Puzzle {
     /** The minimum length for a word to be considered valid and earn points. */
     private final static int MINIMUM_WORD_LENGTH = 4;
-
     /** The number of bonus points recived for finding a pangram. */
     private final static int PANGRAM_BONUS = 7;
 
     /** The primary (required) letter of the puzzle. */
     private char primaryLetter;
-
     /** The secondary letters of the puzzle. */
     private char[] secondaryLetters;
-
+    /** The list of all valid words for the puzzle. */
     private ArrayList<String> validWords;
+    /** The list of all words currently found in the puzzle. */
     private ArrayList<String> foundWords;
+    /** The total number of points that can be earned in the puzzle. */
     private int totalPoints;
+    /** The number of points currently earned in the puzzle. */
     private int earnedPoints;
-    private int[] rankPoints;
+
+    /**
+     * Constructs a Puzzle object from the required letter, and the six other
+     * acceptable letters. Fills validWords by parcing through dictionaryFile.
+     * 
+     * @param primaryLetter The required letter for the puzzle
+     * @param secondaryLetters The six other acceptable letters for the puzzle
+     * @param dictionaryFile The dictionary file to be used to generate validWords.
+     * @throws IOException if an I/O error occurs.
+     */
+    public Puzzle(char primaryLetter, char[] secondaryLetters,
+                  FileReader dictionaryFile) throws IOException {
+        this.primaryLetter = primaryLetter;
+        this.secondaryLetters = Arrays.copyOf(secondaryLetters,
+                                              secondaryLetters.length);
+        this.validWords = new ArrayList<String>();
+        this.foundWords = new ArrayList<String>();
+        this.totalPoints = 0;
+        this.earnedPoints = 0;
+
+        BufferedReader bufferedReader = new BufferedReader(dictionaryFile);
+
+        dictLoop:
+        for (String word = bufferedReader.readLine(); word != null; 
+             word = bufferedReader.readLine()) {
+            if (word.indexOf(primaryLetter) == -1) {
+                continue;
+            }
+
```

Comment:
This for loop will only allow pangrams into the validWords list, as it will continue if any letters are not in the dictionary word being checked. It should instead continue if the word has a letter in it that is not in secondaryLetters. The check for the primary letter is fine because the primary letter is required in all valid words.

Reply:
Sorry, I commented on the wrong lines of code and now they don't show up here. The for loop this is referring to is the loop that checks if all the letters in secondaryLetters appear in the word that it's checking. It should be the very next line of the code that's shown.